src/app/header/auth_store/auth.effects.ts

```diff
@@ -0,0 +1,62 @@
+import { Injectable } from "@angular/core";
+import { Actions, createEffect, ofType } from "@ngrx/effects";
+import { map, switchMap, tap } from "rxjs";
+
+import * as AuthActions from './auth.actions'
+import { HttpClient } from "@angular/common/http";
+import { Router } from "@angular/router";
+
+export interface AuthResponseData {
+    token: string;
+}
+
+@Injectable()
+export class AuthEffects {
+    private apiUrl = 'http://localhost:3004';
+
+    authLogin$ = createEffect(() => 
+        this.actions$.pipe(
+            ofType(AuthActions.Login), 
+            switchMap(authData => {
+                let login = authData.login;
+                let password = authData.password;
+                let tokenOfUSer = '';
+                return this.http.post<AuthResponseData>(
+                    `${this.apiUrl}/auth/login`,
+                    {login, password}
+                )
+                .pipe(
+                    tap(data => {
+                        localStorage.setItem("token", JSON.parse(JSON.stringify(data)).token);
+                        tokenOfUSer = JSON.parse(JSON.stringify(data)).token;;
+                    }),
+                    map(data => {
+                        return JSON.parse(JSON.stringify(data)).token;
+                    }),
```

Comment:
Guess it can be simplified to something like 
`  map(data => {
    const token = JSON.parse(JSON.stringify(data)).token;
    localStorage.setItem('token', token);
    tokenOfUser = token;
    return token;
  })`

Reply:
ok, done